### PR TITLE
Prevent doctrine proxy loading for virtual types

### PR DESCRIPTION
**Pull request by goetas**
BC safe implementation of https://github.com/schmittjoh/serializer/pull/576

Closes #576
Fixes https://github.com/schmittjoh/serializer/issues/575

### CODE REVIEW

**Comment by goetas** [comment]
@scaytrase @maennchen @schmittjoh 

What do you think about this implementation?

I have just added a boolean that allows to decide if proxies should be initialized or not for virtual types

File: src/JMS/Serializer/EventDispatcher/Subscriber/DoctrineProxySubscriber.php [hunk 1]
```diff
@@ -38,6 +48,10 @@ public function onPreSerialize(PreSerializeEvent $event)
         // so it must be loaded if its a real class.
         $virtualType = ! class_exists($type['name'], false);
 
+        if ($this->skipVirtualTypeInit && $virtualType) {
```
Comment (scaytrase) [inline src/JMS/Serializer/EventDispatcher/Subscriber/DoctrineProxySubscriber.php]: I think it's more consistent to put this after `Collection` type overriding

File: tests/JMS/Serializer/Tests/Serializer/EventDispatcher/Subscriber/DoctrineProxySubscriberTest.php [hunk 2]
```diff
@@ -49,6 +49,17 @@ public function testDoesNotRewriteCustomType()
         $this->assertTrue($obj->__isInitialized());
     }
 
+    public function testProxyLoadingCanBeSkippedForVirtualTypes()
```
Comment (scaytrase) [inline tests/JMS/Serializer/Tests/Serializer/EventDispatcher/Subscriber/DoctrineProxySubscriberTest.php]: Now we should have a test with `false` behavior. But I really cannot imagine one.

Comment (goetas) [inline tests/JMS/Serializer/Tests/Serializer/EventDispatcher/Subscriber/DoctrineProxySubscriberTest.php]: this is already done by `testDoesNotRewriteCustomType`

**Pull request merged**